hardware_interface/include/hardware_interface/hardware_info.hpp

```diff
@@ -202,6 +207,27 @@ class HandleDataType
     }
   }
 
+  /**
+   * @brief Cast the given value to double.
+   * @param value The value to be casted.
+   * @return The casted value.
+   * @throw std::runtime_error if the HandleDataType cannot be casted to double.
+   * @note Once we add support for more data types, this function should be updated
+   */
+  double cast_to_double(const HANDLE_DATATYPE & value) const
```

Comment:
I find this good! We will also need this for reuse of controllers.

Also do backwards casting from double to internal values.